misc/python/materialize/checks/owners.py

```diff
@@ -0,0 +1,82 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.util import MzVersion
+
+
+class Owners(Check):
+    def _can_run(self) -> bool:
+        return self.base_version >= MzVersion.parse("0.47.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive("")
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE ROLE owner_role_01;
+                $ postgres-execute connection=postgres://owner_role_01@materialized:6875/materialize
+                CREATE TABLE owner_table1 (c1 int, c2 int)
+                CREATE INDEX owner_index1 ON owner_table1 (c2)
+                CREATE VIEW owner_v1 AS SELECT * FROM owner_table1
+                CREATE MATERIALIZED VIEW owner_mv1 AS SELECT * FROM owner_table1
```

Comment:
`CREATE SOURCE`, even if `GENERATOR` should also be created at some point -- I can bet 100 euro that there are approximately gazillon paths in the code where `CREATE TABLE` and `CREATE SOURCE` are handled differently.

`CREATE SINK` too.

As `CREATE SOURCE` , `CREATE SINK` is expensive, so creating one of each the entire test would be sufficient.